bouncer/shared/gaslimit_ccm.ts

```diff
@@ -119,130 +122,59 @@ async function trackGasLimitSwap(
     ` GasLimit${testTag || ''}`,
   );
 
+  // Do the swap
   const { depositAddress, channelId } = await requestNewSwap(
     sourceAsset,
     destAsset,
     destAddress,
     tag,
     messageMetadata,
   );
-
   const swapRequestedHandle = observeSwapRequested(
     sourceAsset,
     destAsset,
     channelId,
     SwapRequestType.Ccm,
   );
-
-  const reqIdToPrincipalSwapId: { [key: string]: string } = {};
-  const reqIdToGasSwapId: { [key: string]: string } = {};
-  const swapScheduledHandle = observeEvent('swapping:SwapScheduled', {
-    test: (event) => {
-      const data = event.data;
-
-      switch (data.swapType) {
-        case SwapType.CcmPrincipal:
-          reqIdToPrincipalSwapId[data.swapRequestId] = data.swapId;
-          break;
-        case SwapType.CcmGas:
-          reqIdToGasSwapId[data.swapRequestId] = data.swapId;
-          break;
-        default:
-        // not interested in other swap types here
-      }
-      return false;
-    },
-    abortable: true,
-  });
-
-  // SwapExecuted is emitted at the same time as swapScheduled so we can't wait for swapId to be known.
-  const swapIdToEgressAmount: { [key: string]: string } = {};
-  const swapExecutedHandle = observeEvent('swapping:SwapExecuted', {
-    test: (event) => {
-      swapIdToEgressAmount[event.data.swapId] = event.data.outputAmount;
-      return false;
-    },
-    abortable: true,
-  });
-  const swapRequestIdToEgressId: { [key: string]: string } = {};
-  const swapEgressHandle = observeEvent('swapping:SwapEgressScheduled', {
-    test: (event) => {
-      swapRequestIdToEgressId[event.data.swapRequestId] = event.data.egressId;
-      return false;
-    },
-    abortable: true,
-  });
-
-  const egressIdToBroadcastId: { [key: string]: string } = {};
-  const ccmBroadcastHandle = observeEvent(
-    `${destChain.toString().toLowerCase()}IngressEgress:CcmBroadcastRequested`,
-    {
-      test: (event) => {
-        egressIdToBroadcastId[event.data.egressId] = event.data.broadcastId;
-        return false;
-      },
-      abortable: true,
-    },
-  );
-
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const broadcastIdToTxPayload: { [key: string]: any } = {};
-  const broadcastRequesthandle = observeEvent(
-    `${destChain.toString().toLowerCase()}Broadcaster:TransactionBroadcastRequest`,
-    {
+  await send(sourceAsset, depositAddress);
+  const swapRequestId = Number((await swapRequestedHandle).data.swapRequestId.replaceAll(',', ''));
+
+  // Wait for the swap to complete
+  const outputAmount = (
+    await observeEvent('swapping:SwapExecuted', {
+      test: (event) => Number(event.data.swapRequestId.replaceAll(',', '')) === swapRequestId,
+      historicCheckBlocks: CHECK_PAST_BLOCKS_FOR_EVENTS,
```

Comment:
We could probably add a wrapper function that uses `CHECK_PAST_BLOCKS_FOR_EVENTS` (or whatever) by default, so it doesn't have to be repeated each time.

Reply:
doesn't seem worth the extra code.